### ISSUE TITLE
Is there any API to keep the current position always in the center of screen?

### ISSUE THREAD

**Issue by frankgogoland** [mapbox/mapbox-plugins-android#1073]
I had try tracking function (like tracking gps,tracking compass) .I hope there is a mode called follow mode like Baidu Map. Could anyone tell me how to realize it? thanks

**Comment by langsmith** [comment]
What is "follow mode like Baidu Map"?  Try `locationComponent.setCameraMode(CameraMode.TRACKING);`

Camera mode information can be found at https://docs.mapbox.com/android/maps/overview/location-component/#cameramode. 

There's also this demo app example that enables you to see what the different modes are https://docs.mapbox.com/android/maps/examples/location-component-camera-options/

**Comment by frankgogoland** [comment]
Dear Smith:


&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;I had try this API. it is not the real tracking gps. the FOLLOW mode means the current gps is always in the center of screen.&nbsp;


&nbsp;thanks




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Langston Smith"<notifications@github.com&gt;;
发送时间:&nbsp;2019年11月24日(星期天) 上午7:47
收件人:&nbsp;"mapbox/mapbox-plugins-android"<mapbox-plugins-android@noreply.github.com&gt;;
抄送:&nbsp;"冰冻三尺"<1223774014@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [mapbox/mapbox-plugins-android] Is there any API to keep the current position always in the center of screen? (#1073)




What is "follow mode like Baidu Map"?  Try locationComponent.setCameraMode(CameraMode.TRACKING);
 
Camera mode information can be found at https://docs.mapbox.com/android/maps/overview/location-component/#cameramode.
 
There's also this demo app example that enables you to see what the different modes are https://docs.mapbox.com/android/maps/examples/location-component-camera-options/

**Comment by sebastienrouif** [comment]
> the FOLLOW mode means the current gps is always in the center of screen
do you mean by that that the user should not be able to move the map away?

**Comment by frankgogoland** [comment]
NO, user can move the other place with select normal mode. I suggest you download a baidmap demo .


BTW, Because abroad customer hope me to use Mapbox,so I really hope you can optimize this function.&nbsp;


I use CircleManager create function to draw circle every 1 seconds. when the points arrive to 3000 points , I can move map any more.


I don't know reason.






------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Sebastien ROUIF"<notifications@github.com&gt;;
发送时间:&nbsp;2019年11月25日(星期一) 晚上10:39
收件人:&nbsp;"mapbox/mapbox-plugins-android"<mapbox-plugins-android@noreply.github.com&gt;;
抄送:&nbsp;"冰冻三尺"<1223774014@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [mapbox/mapbox-plugins-android] Is there any API to keep the current position always in the center of screen? (#1073)



 
the FOLLOW mode means the current gps is always in the center of screen
 do you mean by that that the user should not be able to move the map away?

**Comment by langsmith** [comment]
> download a baidmap demo

Can you share a link so that I can download this demo? I still don't understand what the difference will be between `CameraMode.TRACKING` and Baidu's follow mode.

**Comment by frankgogoland** [comment]
Yes, you can download my product in CoolMarket.


https://www.coolapk.com/apk/make.more.r2d2.cellular_pro


App will prompt&nbsp;&nbsp;if your phone is not root.warning, please ignore it









thanks






------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Langston Smith"<notifications@github.com&gt;;
发送时间:&nbsp;2019年11月26日(星期二) 凌晨0:20
收件人:&nbsp;"mapbox/mapbox-plugins-android"<mapbox-plugins-android@noreply.github.com&gt;;
抄送:&nbsp;"冰冻三尺"<1223774014@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [mapbox/mapbox-plugins-android] Is there any API to keep the current position always in the center of screen? (#1073)



 
download a baidmap demo
  
Can you share a link so that I can download this demo? I still don't understand what the difference will be between CameraMode.TRACKING and Baidu's follow mode.
